### PR TITLE
Remove assignments to Child/Children in OsuClickableContainer subclasses

### LINKED ISSUE #21676: `LeaderboardScore` and some other `OsuClickableContainer`s do not have hover (click) sounds
### Type

Game behaviour

### Bug description

Regressed in https://github.com/ppy/osu/pull/21619.

That PR made hover sounds possible to be removed when using `Child`/`Children` in `load()`/`LoadComplete()`. Not sure of a good solution other than:
- Replacing those with `Add()`/`AddRange()` and restricting `Child`/`Children` somehow
- Moving `CreateHoverSounds()` to `LoadComplete()` and hope there's no `Child`/`Children` calls after

### Screenshots or videos

https://user-images.githubusercontent.com/35318437/208231081-eb73a1ff-6662-4bb9-bcfe-a510ade9db3e.mp4

### Version

master

### Logs

N/A

### PR DESCRIPTION
**Pull request by kemley76**
Fixes #21676
There may still be some instances where `Clear()` is called on `OsuClickableContainer` from assigning to `Child` or `Children`, but I removed as many as I could find.

### CODE REVIEW

**Comment by bdach** [comment]
Please check code inspection and test failures.

**Comment by bdach** [comment]
Code quality still failing.

File: osu.Game.Tests/Visual/Multiplayer/QueueModeTestScene.cs [hunk 1]
```diff
@@ -86,7 +85,6 @@ public override void SetUpSteps()
             ClickButtonWhenEnabled<MultiplayerMatchSettingsOverlay.CreateOrUpdateButton>();
 
             AddUntilStep("wait for join", () => MultiplayerClient.RoomJoined);
-            AddUntilStep("wait for ongoing operation to complete", () => !(CurrentScreen as OnlinePlayScreen).ChildrenOfType<OngoingOperationTracker>().Single().InProgress.Value);
```
Comment (bdach) [inline osu.Game.Tests/Visual/Multiplayer/QueueModeTestScene.cs:89]: Why is this in the diff? It looks like this pull is reverting https://github.com/ppy/osu/pull/22450 for whatever reason?

Please fix the history here. Probably easiest to rebase your work onto current `master` and force-push to fix (this is a special exemption, do *not* do this normally).

Comment (wiskerz76) [reply osu.Game.Tests/Visual/Multiplayer/QueueModeTestScene.cs]: Sorry about that. It seems that when trying the rebase my changes onto the latest master I had gotten some of the newer changes caught up in my commit. I believe I have now fixed the issue.

File: osu.Game.Tests/Visual/Online/TestSceneBeatmapListingOverlay.cs [hunk 2]
```diff
@@ -48,7 +48,8 @@ public void SetUpSteps()
         {
             AddStep("setup overlay", () =>
             {
-                Child = overlay = new BeatmapListingOverlay { State = { Value = Visibility.Visible } };
```
Comment (peppy) [inline osu.Game.Tests/Visual/Online/TestSceneBeatmapListingOverlay.cs:51]: This looks unrelated?

File: osu.Game.Tests/Visual/UserInterface/TestSceneOsuHoverContainer.cs [hunk 3]
```diff
@@ -22,17 +22,19 @@ public partial class TestSceneOsuHoverContainer : OsuManualInputManagerTestScene
...
-                {
-                    RelativeSizeAxes = Axes.Both,
-                },
             };
+            colourContainer = new Box
+            {
+                RelativeSizeAxes = Axes.Both,
+            };
+            hoverContainer.Add(colourContainer);
+            Add(hoverContainer);
```
Comment (peppy) [inline osu.Game.Tests/Visual/UserInterface/TestSceneOsuHoverContainer.cs:37]: This is shocking readability wise.

**Comment by peppy** [comment]
I disagree with this on a conceptual level. `Clear` should never be disabled for a reason like this.

Needs a different solution.

**Review by peppy — COMMENTED** [review]
Unfinished review because I realised that this whole direction is wrong halfway through.
